### PR TITLE
simulators/ethereum: Update beacon root address 0x000F..Beac02

### PR DESCRIPTION
**Pull request by danceratopz**
Updates to the final EIP-4788 Beacon Root Address, cf https://github.com/ethereum/execution-spec-tests/pull/312.

### CODE REVIEW

File: simulators/ethereum/engine/config/cancun/constants.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ var (
 	BLOB_COMMITMENT_VERSION_KZG = byte(0x01)
 
 	// EIP 4788
-	BEACON_ROOTS_ADDRESS  = common.HexToAddress("0xbEAC020008aFF7331c0A389CB2AAb67597567d7a")
+	BEACON_ROOTS_ADDRESS  = common.HexToAddress("0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02")
 	HISTORY_BUFFER_LENGTH = uint64(98304)
```
Comment (spencer-tb) [inline simulators/ethereum/engine/config/cancun/constants.go:26]:
```suggestion
	HISTORY_BUFFER_LENGTH = uint64(8191)
```

**Review by spencer-tb — CHANGES_REQUESTED** [review]
Nice - we need to update the history buffer length and also the code in `simulators/ethereum/engine/config/cancun/genesis.go` to:
```
3373fffffffffffffffffffffffffffffffffffffffe14604d57602036146024575f5ffd5b5f35801560495762001fff810690815414603c575f5ffd5b62001fff01545f5260205ff35b5f5ffd5b62001fff42064281555f359062001fff015500
```

**Review by marioevz — APPROVED** [review]
LGTM! :)

**Review by spencer-tb — APPROVED** [review]
LGTM :D

**Pull request merged**